File: Native/Test/Python/mr-array-cache.py
import fabric

client = fabric.createClient()

cv = client.MR.createConstValue("Size", 10)

ago = client.KLC.createArrayGeneratorOperator("foo.kl", "operator foo(io Scalar output, Size index) { report 'Running generator'; output = sqrt(Scalar(index)); }", "foo");
ag = client.MR.createArrayGenerator(cv, ago)

cache = client.MR.createArrayCache(ag)

count = ag.getCount()
print("cache.getCount() = " + str(count))

print("Should run generator")
for i in range(0, count):
  print("cache.produce("+str(i)+") = "+str(cache.produce(i)))

print("Should not run generator (cached)")
for i in range(0, count):
  print("cache.produce("+str(i)+") = "+str(cache.produce(i)))

cache.flush()
print("Should run generator")
for i in range(0, count):
  print("cache.produce("+str(i)+") = "+str(cache.produce(i)))

client.close()

File: Native/Test/Node/mr-array-cache.js
FC = require("Fabric").createClient();

cv = FC.MR.createConstValue("Size", 10);

ago = FC.KLC.createArrayGeneratorOperator("foo.kl", "operator foo(io Scalar output, Size index) { report 'Running generator'; output = sqrt(Scalar(index)); }", "foo");
ag = FC.MR.createArrayGenerator(cv, ago);

var cache = FC.MR.createArrayCache(ag);

var count = ag.getCount();
console.log("cache.getCount() = " + count);

console.log("Should run generator");
for (var i=0; i<count; ++i)
  console.log("cache.produce("+i+") = "+cache.produce(i));

console.log("Should not run generator (cached)");
for (var i=0; i<count; ++i)
  console.log("cache.produce("+i+") = "+cache.produce(i));

cache.flush();
console.log("Should run generator");
for (var i=0; i<count; ++i)
  console.log("cache.produce("+i+") = "+cache.produce(i));

FC.close();
